src/net/dmytro/javapro/homework5/task2/obstacle/Course.java

```diff
@@ -0,0 +1,16 @@
+package net.dmytro.javapro.homework5.task2.obstacle;
+
+import net.dmytro.javapro.homework5.task2.Team;
+
+public class Course {
+    private Obstacle[] obstacles;
+
+    public Course(Obstacle ... obstacles) {
+        this.obstacles = obstacles;
+    }
+    public void doIt(Team team){
```

Comment:
I dont think you need that class. it changes logic ... 

in the task participant have to pass set of obstacles, and if he failed he dont start new one. 

by this method whole team is trying to pass same obstacle, and in case some failed he start new. 

I would remove this method, and updated doIt() method in Team class.